src/wp-includes/blocks.php

```diff
@@ -1870,8 +1870,16 @@ function resolve_pattern_blocks( $blocks ) {
 				continue;
 			}
 
-			$blocks_to_insert   = parse_blocks( $pattern['content'] );
-			$seen_refs[ $slug ] = true;
+			$blocks_to_insert = parse_blocks( trim( $pattern['content'] ) );
+
+			// For single-root patterns, add the pattern name to make this a pattern instance in the editor.
+			if ( count( $blocks_to_insert ) === 1 ) {
+				$blocks_to_insert[0]['attrs']['metadata'] = array(
+					'patternName' => $slug,
+					'name'        => $pattern['title'],
+				);
```

Comment:
Interesting. Something in this branch prevents the "recursive pattern" test ([below](https://github.com/WordPress/wordpress-develop/pull/10248/files#diff-dbadfafed69b4bfd8648cc81be3538207d23519e3e96d8337552a530252291e4R92)) from running at all. The PHP unit tests just stop and time out with no error.